Assets/_Project/Scripts/Navigation/NavigationPointProvider.cs

```diff
@@ -0,0 +1,24 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using Mapbox.Unity.Map;
+using Mapbox.Unity.Utilities;
+using Mapbox.Utils;
+using UnityEngine;
+
+namespace PolSl.UrbanHealthPath
+{
+    public class NavigationPointProvider : MonoBehaviour
```

Comment:
What does it provide? I think it needs a complete rework. Maybe call it "PositionIndicatorRefresher", turn the method to UpdateIndicator(Vector3d position), create a class responsible for fetching the location data (TimedLocationFetcher) that would have LocationFetched event and subscribe PositionIndicatorRefresher to that event? Or something similiar.

Reply:
Nevermind, I just realized that it does completely different thing and is used for creating navigation path. TBH I don't think it's needed, we should consider a different approach (maybe NavigationManager?), but I'd leave it for now and refactor this functionality on a separate branch.